aibolit/patterns/empty_rethrow/empty_rethrow.py

```diff
@@ -1,47 +1,27 @@
-# The MIT License (MIT)
-#
-# Copyright (c) 2020 Aibolit
-#
-# Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and associated documentation files (the "Software"), to deal
-# in the Software without restriction, including without limitation the rights
-# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-# copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included
-# in all copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-# SOFTWARE.
-
-import javalang
-
+from aibolit.ast_framework import ASTNodeType, AST
 from aibolit.utils.ast_builder import build_ast
+from typing import List, Set
 
 
 class EmptyRethrow:
 
     def __init__(self):
         pass
 
-    def value(self, filename):
-        tree = build_ast(filename)
-        total_code_lines = set()
-        for _, method_node in tree.filter(javalang.tree.MethodDeclaration):
-            for _, try_node in method_node.filter(javalang.tree.TryStatement):
-                for _, throw_node in try_node.filter(javalang.tree.ThrowStatement):
-                    if try_node.catches:
-                        catch_classes = [x.parameter.name for x in try_node.catches]
-                        mem_ref = throw_node.children[1]
-                        if isinstance(mem_ref, javalang.tree.ClassCreator):
+    def value(self, filename) -> List[int]:
+        total_code_lines: Set = set()
+        ast = AST.build_from_javalang(build_ast(filename))
+        method_decls = list(ast.get_proxy_nodes(ASTNodeType.METHOD_DECLARATION))
+        for method_node in method_decls:
+            for try_node in list(ast.get_proxy_nodes(ASTNodeType.TRY_STATEMENT)):
+                for throw_node in list(ast.get_proxy_nodes(ASTNodeType.THROW_STATEMENT)):
+                    field_catche = try_node.catches
+                    if field_catche:
+                        catch_classes = [x.parameter.name for x in field_catche]
+                        mem_ref = list(throw_node.children)[0]
+                        if mem_ref.node_type == ASTNodeType.CLASS_CREATOR:
                             continue
                         else:
                             if hasattr(mem_ref, 'member') and mem_ref.member in catch_classes:
```

Comment:
```suggestion
                        if mem_ref.node_type == ASTNodeType.CLASS_CREATOR and \
                           hasattr(mem_ref, 'member') and mem_ref.member in catch_classes:
```
Also adjust next line.

Reply:
Can we omit the use of `hasattr`?

Reply:
nope we can not. We want to check childs ASTNodeType.THROW_STATEMENT. Imperically it was devided to two cases: if we met CLASS_CREATOR and other objects. Not every child has this attribute, that is why we need to check it.